apps/api/src/services/email-verification.service.ts

```diff
@@ -0,0 +1,199 @@
+import axios from "axios";
+
+import { config } from "@/config/index.js";
+import { findById as findAgentById } from "@/database/repositories/agent-repository.js";
+import {
+  findEmailVerificationTokenByToken,
+  markTokenAsUsed,
+} from "@/database/repositories/email-verification-repository.js";
+import { findById as findUserById } from "@/database/repositories/user-repository.js";
+import { AgentManager } from "@/services/agent-manager.service.js";
+import { UserManager } from "@/services/user-manager.service.js";
+
+type Result<Ok, Err> =
```

Comment:
in general, our return types and errors in this codebase are messy. some exist in the root `types`; some are within a file; some are in the database defs. 

although this rust-like pattern could be generally useful, it's not used anywhere else.

that said, https://github.com/recallnet/js-recall/issues/632 needs to happen and can unify a consistent pattern across each of our app layers. this might be a useful first step in that direction, so i'm fine if we want to keep this logic.

i know @joewagner has some thoughts on error patterns, types, and zod, so it might help to get his thoughts here.